### PR TITLE
Feature/query error handling krd

### LINKED ISSUE #22: Add more error handling for network requests in server.js

### PR DESCRIPTION
**Pull request by krdurrett**
## What Features are Being Added?
- Create error handling in get request for specific book
- Create error handling in post request for favorites
- Create error handling in delete request for favorites

## What is Being Refactored?
- Refactored delete request in queries (took out .then)

## Where should the reviewer start?
- Queries.js then server.js

## How should this be manually tested?
- Postman

## What are the relevant tickets?
- n/a

## Any Other Notes/Questions/Problems?
-Note for Robbie:
We started with creating our Express server. We sorted through the NYT bestseller book data that we obtained through the NYT API. We cleaned the data to make it more readable/applicable for the scope of this project.
We then built out the basic functionality of an Express server with the network requests we will be using in our application (GET, POST, DELETE), simply using the app.locals Express interface. Once we had our local Express database set up with all request functionality working, we moved on to creating a PostgreSQL database. This required us to begin our learning on PostgreSQL, knex.js, and database configuration in general. We utilized many resources, including Knex.js documentation, to understand how to migrate and seed our tables within our PG database. Once we had successfully set up the database using migrations and seeding, we could then move on to refactoring our server file to run queries to our PG database. We created queries to obtain all books or one specific book from our main book data table, and also queries to add/delete/and obtain data from our other table, book favorites. Once those were written we linked them to our GET/POST/DELETE requests in the server.js file. We tested these queries/network requests using Postman. Lastly, here we have added some error handling (not sure if that is what its called on the backend) to inform users of errors obtaining data.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -24,8 +20,16 @@ app.get('/api/v1/books', (request, response) => {
 
 app.get('/api/v1/books/:id', (request, response) => {
   queries.getSingleBook(request)
-    .then(data => response.status(200).json(data))
-    .catch(error => response.status(404).json({ error }));
+    .then(books => {
```
Comment (Fordo29) [inline server.js:23]: Well done on the error handling for this

File: server.js [hunk 2]
```diff
@@ -49,10 +53,16 @@ app.post('/api/v1/favorites', (request, response) => {
 });
 
 app.delete('/api/v1/favorites/:isbn', (request, response) => {
-  queries.removeBookFromFavs(request);
-  response.status(200).json({
-    message: `Book with isbn #${request.params.isbn} has been deleted from favorites`
-  })
+  queries.removeBookFromFavs(request)
+    .then(count => {
+      if (count) {
```
Comment (cagallo) [inline server.js:58]: Great logic! Need to target the count (count of deleted rows- i.e. 1 if you delete a book) vs the favorites books array itself.

Comment (Fordo29) [inline server.js:58]: Good find that the del() returns a number of how many rows were deleted.

File: server.js (hunk 1 above)
Comment (cagallo) [inline server.js:23]: Good to add the conditional here!

**Review by cagallo — APPROVED** [review]

**Comment by ectweitmann** [comment]
Nice work! We're making great progress!

File: db/seeds/books_favorites.js [hunk 3]
```diff
@@ -19,8 +19,5 @@ exports.seed = function(knex) {
       .then(() => {
         return knex.raw('ALTER SEQUENCE favorites_id_seq RESTART WITH 1');
```
Comment (robbiejaeger) [inline db/seeds/books_favorites.js:20]: Did you find that you needed to reset the primary key when re-seeding? Usually, this isn't necessary and it's ok to let the primary key keep incrementing. But was something relying on the primary keys to start over?

**Comment by robbiejaeger** [comment]
I added a comment that was more of a question, but the rest of the code looks good.

**Pull request merged**